File: frontend/src/services/FlightService.js
import axios from "axios";

const API_URL = "http://localhost:8080/flights";

class FlightService {

    getAllFlights() {
        return axios.get(API_URL);
    }

    saveFlight(flight) {
        return axios.post(API_URL, flight);
    }

    deleteFlight(code) {
        return axios.delete(`${API_URL}/${code}`);
    }

    searchByCarrier(carrier) {
        return axios.get(`${API_URL}/carrier/${carrier}`);
    }

    searchByRoute(source, destination) {
        return axios.get(
            `${API_URL}/route?source=${source}&destination=${destination}`
        );
    }

    searchByPrice(min, max) {
        return axios.get(
            `${API_URL}/price?min=${min}&max=${max}`
        );
    }
}

export default new FlightService();

File: flightservice/src/main/java/com/flight/service/FlightService.java
package com.flight.service;

import java.util.List;
import com.flight.entity.Flight;

public interface FlightService {

    Flight save(Flight flight);

    Flight findByCode(String code);

    List<Flight> findByCarrier(String carrier);

    List<Flight> findByRoute(String source, String destination);

    List<Flight> findByPriceRange(Double min, Double max);

    List<Flight> list();

    void delete(String code);
}